agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -153,12 +153,62 @@ def ensure_terminated
     end
 
     # @param [Docker::Container] service_container
-    # @param [Kontena::Models::ServicePod] service_pod
-    def service_container_outdated?(service_container, service_pod)
-      creator = Kontena::ServicePods::Creator.new(service_pod)
-      creator.container_outdated?(service_container) ||
-        creator.labels_outdated?(service_pod.labels, service_container) ||
-          creator.recreate_service_container?(service_container)
+    def service_container_outdated?(service_container)
+      outdated = container_outdated?(service_container) ||
+          labels_outdated?(service_container) ||
+          recreate_service_container?(service_container)
+      return true if outdated
+
+      image_puller.ensure_image(
```

Comment:
Now that the `ServicePodWorker` handles the image re-pull on update checks, the `ServicePods::Creator` probably doesn't need to pull it twice anymore. It does still need to pull it, because the `config_container` -> `Weave` stuff needs access to the image before creating the container. Perhaps the `ServicePods::Creator#create_container` -> `ensure_image` call could be removed?